modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java

```diff
@@ -123,7 +129,39 @@ public static void main(String[] args) throws IOException {
         return contextInfos;
     }
 
-    private static Set<PainlessContextClassInfo> createShared(List<PainlessContextInfo> contextInfos) {
+    private static Set<Object> createSharedStatics(List<PainlessContextInfo> contextInfos) {
+        Map<Object, Integer> staticInfoCounts = new HashMap<>();
+
+        for (PainlessContextInfo contextInfo : contextInfos) {
+            for (PainlessContextMethodInfo methodInfo : contextInfo.getImportedMethods()) {
+                staticInfoCounts.compute(methodInfo, (k, v) -> v == null ? 1 : v + 1);
```

Comment:
I think this can be done more simply with `merge`:

```
staticInfoCounts.merge(methodInfo, 1, Integer::sum)
```

Reply:
Good catch.  Fixed.